src/main/java/com/microsoft/jenkins/azuread/integrations/casc/AzureAdMatrixAuthorizationStrategyConfigurator.java

```diff
@@ -0,0 +1,89 @@
+package com.microsoft.jenkins.azuread.integrations.casc;
+
+import com.microsoft.jenkins.azuread.AzureAdMatrixAuthorizationStrategy;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.Permission;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.impl.attributes.MultivaluedAttribute;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import org.jenkinsci.plugins.matrixauth.AuthorizationContainer;
+import org.jenkinsci.plugins.matrixauth.integrations.PermissionFinder;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Extension(optional = true)
+@Restricted(NoExternalUse.class)
+public class AzureAdMatrixAuthorizationStrategyConfigurator extends
+        BaseConfigurator<AzureAdMatrixAuthorizationStrategy> {
+    @NonNull
+    @Override
+    public String getName() {
+        return "azureAdMatrix";
+    }
+
+    @Override
+    public Class<AzureAdMatrixAuthorizationStrategy> getTarget() {
+        return AzureAdMatrixAuthorizationStrategy.class;
+    }
+
+    @NonNull
+    @Override
+    public Class getImplementedAPI() {
+        return AuthorizationStrategy.class;
+    }
+
+    @Override
+    @NonNull
+    public Set<Attribute<AzureAdMatrixAuthorizationStrategy, ?>> describe() {
+        return new HashSet<>(Collections.singletonList(
+                new MultivaluedAttribute<AzureAdMatrixAuthorizationStrategy, String>("permissions", String.class)
+                        .getter(AzureAdMatrixAuthorizationStrategyConfigurator::getPermissions)
+                        .setter(AzureAdMatrixAuthorizationStrategyConfigurator::setPermissions)
+        ));
+    }
+
+    @Override
+    protected AzureAdMatrixAuthorizationStrategy instance(Mapping mapping, ConfigurationContext context) {
+        return new AzureAdMatrixAuthorizationStrategy();
+    }
+
+    @Override
+    public CNode describe(AzureAdMatrixAuthorizationStrategy instance, ConfigurationContext context) throws Exception {
+        return compare(instance, new AzureAdMatrixAuthorizationStrategy(), context);
+    }
+
+    /**
+     * Extract container's permissions as a List of "PERMISSION:sid".
+     */
+    public static Collection<String> getPermissions(AuthorizationContainer container) {
```

Comment:
can't you reuse the code from matrix-auth?
https://github.com/jenkinsci/matrix-auth-plugin/blob/master/src/main/java/org/jenkinsci/plugins/matrixauth/integrations/casc/MatrixAuthorizationStrategyConfigurator.java#L50

Reply:
Yes, should just extend from `MatrixAuthorizationStrategyConfigurator` class here, just to add log for debug temporarily.